api/transport/header.go

```diff
@@ -102,6 +111,20 @@ func (h Headers) Items() map[string]string {
 	return h.items
 }
 
+// ForwardingItems returns the non-canonicalized version of the underlying map
+// for this Headers object. The returned map MUST NOT be changed.
+// Doing so will result in undefined behavior.
```

Comment:
You're building a new map. It's safe to mutate the result of this function. You'll want to skip the `emptyMap` thing.